tsdb/head.go

```diff
@@ -1211,6 +1220,14 @@ func (a *headAppender) AddFast(ref uint64, t int64, v float64) error {
 	return nil
 }
 
+func (a *headAppender) AddExemplar(lset labels.Labels, t int64, e exemplar.Exemplar) error {
+	return nil
+}
+
+func (a *headAppender) AddExemplarFast(ref uint64, t int64, v float64, e exemplar.Exemplar) error {
+	return nil
+}
+
```

Comment:
Looking at the WAL PR and the empty methods here, I am wondering if the examplar storage should be a part of Head itself and not something which is handled separately. Else there seems to some disconnect between the WAL and the exemplarStorage used separately. The WAL write should happen first, and only after its success we add to the in-memory storage.

Reply:
And with that, we don't need to store the labels in the exemplar and maybe cause possible duplication of string, and just store the reference of the series it is related to.

Reply:
Additionally, the exemplar should go in the appender buffer at first and only go in the storage once we call Commit().

Reply:
@codesome could still use another review re: the changes to head.go